extract_unique_sensors_01.py

```diff
@@ -86,10 +86,10 @@ def extract_all_sensors(sensors_data, sensor_name):
     assert isinstance(sensor_name, str), "Expected a string."
     assert isinstance(sensors_data, list), "Expected a list."
     candidate_lines = []
-    for sensor_data in sensors_data:
-        if sensor_data.startswith(sensor_name):
-            for candidate_line in sensor_data.split():
-                candidate_lines.append(candidate_line)
+    candidate_lines = [
+        candidate_line for sensor_data in sensors_data if sensor_data.startswith(sensor_name)
+        for candidate_line in sensor_data.split()
```

Comment:
You could review the comments in the mattermost thread: https://kunye.sarao.ac.za/cam/pl/rabxf3qr37r3zgcjhn7snbmy7r wrt how you would implement list comprehension, you are on the right path.
Also, I would advice you to consider writing the tests first - should you need assistance ping either @shoosen or myself.

Reply:
@mmphego  Oh I see, great talk by Conor Hoekstra and I get the picture now on when to use list comprehensions(nicely explained in mattermost comments).